New apps are being generated with old dependencies

**What is the problem?**

Newly generated apps are being populated with the *oldest* version of dependencies available in the major version range specified in the template rather than the newest.

**Steps to Reproduce:**

1. `blitz new my-blitz-project`
1. `cd my-blitz-project`
1. `yarn lint` 💥 Oops,

`eslint-plugin-react-hooks@2.0.0` is broken. Wait, why do we have 2.0.0 anyway? 2.5.1 is the latest 2.x release.

**Versions:**

- Blitz: 0.6.0.-canary.0
- OS: Windows 10 and macOS

```
debug: local
debug: pkgPath: C:\dev\test\my-blitz\node_modules\@blitzjs\cli

env: win32-x64 node-v12.16.1
blitz: 0.6.0-canary.0 (global)
blitz: 0.6.0-canary.0 (local)
```

**Supporting Documentation**

```
> blitz new my-broken-blitz
> cd my-broken-blitz
> yarn lint

$ eslint --ignore-path .gitignore --ext .js,.ts,.tsx .
Error: Failed to load plugin 'react-hooks' declared in '.eslintrc.js » eslint-config-react-app': Cannot find module './src/index'
Require stack:
- C:\dev\test\my-broken-blitz\node_modules\eslint-plugin-react-hooks\index.js
- C:\dev\test\my-broken-blitz\node_modules\eslint\lib\cli-engine\config-array-factory.js
- C:\dev\test\my-broken-blitz\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js
- C:\dev\test\my-broken-blitz\node_modules\eslint\lib\cli-engine\cli-engine.js
- C:\dev\test\my-broken-blitz\node_modules\eslint\lib\cli-engine\index.js
- C:\dev\test\my-broken-blitz\node_modules\eslint\lib\cli.js
- C:\dev\test\my-broken-blitz\node_modules\eslint\bin\eslint.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:982:15)
    at Function.Module._load (internal/modules/cjs/loader.js:864:27)
    at Module.require (internal/modules/cjs/loader.js:1044:19)
    at require (C:\dev\test\my-broken-blitz\node_modules\v8-compile-cache\v8-compile-cache.js:161:20)
    at Object.<anonymous> (C:\dev\test\my-broken-blitz\node_modules\eslint-plugin-react-hooks\index.js:10:18)
    at Module._compile (C:\dev\test\my-broken-blitz\node_modules\v8-compile-cache\v8-compile-cache.js:194:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
    at Function.Module._load (internal/modules/cjs/loader.js:901:14)
    at Module.require (internal/modules/cjs/loader.js:1044:19)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

> yarn outdated

Package                   Current                      Wanted                       Latest  Package Type    URL
eslint-plugin-react-hooks 2.0.0                        2.0.0                        3.0.0   devDependencies https://reactjs.org/
```

--- a/packages/cli/src/utils/get-latest-version.ts
+++ b/packages/cli/src/utils/get-latest-version.ts
@@ -11,7 +11,8 @@ export const getLatestVersion = async (dependency: string, templateVersion: stri
 
   const latestVersion = Object.keys(allVersions)
     .filter((version) => version.startsWith(major))
-    .sort((a, b) => a.localeCompare(b, undefined, {numeric: true}))[0]
+    .sort((a, b) => a.localeCompare(b, undefined, {numeric: true}))
+    .reverse()[0]
 
   // If the latest tagged version matches our pinned major, use that, otherwise use the
   // latest untagged which does

